### ISSUE TITLE
HTTPS Authentication not working on Linux

### ISSUE THREAD

**Issue by edudobay** [Microsoft/vscode#1796]
On VS Code 0.10.6, when trying to sync changes in a HTTPS-backed Git repository, I get an error message which expands to [1] when I click 'show output', if VS Code was run from the desktop environment. If I close VS Code and run it from the terminal instead, I get the username and password prompts *on the terminal* and typing from there just works (although it's obviously an uncomfortable task-switch).

[1]
```
git fetch
Missing ipc hook
error: unable to read askpass response from '$APPDIR/resources/app/out/vs/workbench/parts/git/electron-main/askpass.sh'
fatal: could not read Username for 'https://bitbucket.org': No such device or address
git pull
Missing ipc hook
error: unable to read askpass response from '$APPDIR/resources/app/out/vs/workbench/parts/git/electron-main/askpass.sh'
fatal: could not read Username for 'https://bitbucket.org': No such device or address
```
(Note: for conciseness I've replaced the directory where the application is installed with $APPDIR)

And here's what I see in the developer tools console:
```
workbench.main.js:63 There was an issue running a git operation.
Please review the output or use a console to check the state of
your repository.: Error: There was an issue running a git operation.
Please review the output or use a console to check the state of
your repository.
    at Object.S [as create] (file:///$APPDIR/resources/app/out/vs/workbench/workbench.main.js:10:2854)
    at file:///$APPDIR/resources/app/out/vs/workbench/workbench.main.js:62:16688
    at Object.p [as _notify] (file:///$APPDIR/resources/app/out/vs/workbench/workbench.main.js:6:181)
    at Object.__dirname.undefined.F.enter (file:///$APPDIR/resources/app/out/vs/workbench/workbench.main.js:6:3509)
    at __dirname.undefined.t.Class.define._run (file:///$APPDIR/resources/app/out/vs/workbench/workbench.main.js:6:4830)
    at __dirname.undefined.t.Class.define._chainedError (file:///$APPDIR/resources/app/out/vs/workbench/workbench.main.js:6:4189)
    at n (file:///$APPDIR/resources/app/out/vs/workbench/workbench.main.js:6:2017)
    at Object.p [as _notify] (file:///$APPDIR/resources/app/out/vs/workbench/workbench.main.js:6:181)
    at Object.__dirname.undefined.F.enter (file:///$APPDIR/resources/app/out/vs/workbench/workbench.main.js:6:3509)
    at __dirname.undefined.t.Class.define._run (file:///$APPDIR/resources/app/out/vs/workbench/workbench.main.js:6:4830)e.doShow @ workbench.main.js:63
```

And some more version info:
```
Version 0.10.6
Commit a80232bbcfe8a5cdad1ebc98638673d9dcb02458
Date 2015-12-19T11:26:42.359Z
Shell 0.34.5
Renderer 45.0.2454.85
Node 4.1.1
```

**Comment by joaomoreno** [comment]
What OS are you on?

**Comment by edudobay** [comment]
I'm running openSUSE Tumbleweed, 64 bits. Do I need to be more specific?

**Comment by joaomoreno** [comment]
No need, that's enough. This is **always** reproducible? Given the error message, other things should be crashing down...

**Comment by edudobay** [comment]
Yes, whenever Git has to HTTP-authenticate with the remote repo. The same bug happens on Debian Jessie. Up to now, other Git features seem to be working. Remote operations with SSH key authentication do work.

**Comment by anudeep3998** [comment]
Hi. I have a similar issue. 
Authentication fails, always. It seems to be connecting as an anon user. The log output is
`remote: Anonymous access to xxx failed. `
This is after entering the username and password in the corresponding popup that shows up when trying to sync.

**Comment by datimson** [comment]
I have the same "Missing ipc hook" error on Ubuntu 14.04 LTS with git 2.7. I'm trying to connect to my visual studio online repository which uses HTTPS.

**Comment by melo0187** [comment]
I can also see this bug. All the interactions with the git remote, like the git fetch when opening the project in vsc or a manual git pull from within vsc give the "Missing ipc hook" error. Since I provide the user in the URL, my output looks like this:
```
git pull
Missing ipc hook
error: unable to read askpass response from '/home/melo/Downloads/Apps/VSCode-linux-x64/resources/app/out/vs/workbench/parts/git/electron-main/askpass.sh'
fatal: could not read Password for 'https://user@my.gitblit.server:8443': No such device or address
```

**Comment by busbina** [comment]
I am getting this in Windows 10 with the newest update to code.

**Comment by joaomoreno** [comment]
Pushing to April.

Please consider using a [credential store](https://help.github.com/articles/caching-your-github-password-in-git/) as a temporary workaround.

**Comment by joaomoreno** [comment]
Fixed with 4849118632d44bff916550e2babb492e1cca9646
